File: DeviantArtFs/Types/ResponseTypes/StashDeltaResult.fs
namespace DeviantArtFs

open System
open FSharp.Json

type StashDeltaResult = {
    cursor: string
    has_more: bool
    next_offset: int option
    reset: bool
    entries: StashDeltaEntry list
} with
    static member Parse json = Json.deserialize<StashDeltaResult> json
    member this.GetNextOffset() = OptUtils.intDefault this.next_offset
    interface IResultPage<int, StashDeltaEntry> with
        member this.HasMore = this.has_more
        member this.Cursor = this.next_offset |> Option.defaultValue 0
        member this.Items = this.entries |> Seq.ofList

File: DeviantArtFs.Types/StashDeltaResult.cs
using System;
using System.Collections.Generic;

namespace DeviantArtFs.Types {
    public class StashDeltaResult : IResultPage<int, StashDeltaEntry> {
        public string cursor { get; set; } = "";
        public bool has_more { get; set; }
        public int? next_offset { get; set; }
        public bool reset { get; set; }
        public StashDeltaEntry[] entries { get; set; } = new StashDeltaEntry[0];

        bool IResultPage<int, StashDeltaEntry>.HasMore => has_more;
        int IResultPage<int, StashDeltaEntry>.Cursor => next_offset ?? 0;
        IEnumerable<StashDeltaEntry> IResultPage<int, StashDeltaEntry>.Items => entries;
    }
}
